modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/SlaCard/index.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import ClayIcon from '@clayui/icon';
+import classNames from 'classnames';
+import {useEffect, useState} from 'react';
+import {FORMAT_DATE} from '../../../../common/utils/constants/slaCardDate';
+import {SLA_NAMES} from '../../../../common/utils/constants/slaCardNames';
+import SlaCardLayout from './Layout';
+
+const SlaCard = ({project}) => {
+	const [slaData, setSlaData] = useState();
+	const [slaSelected, setSlaSelected] = useState('');
+	const SLA_DATA = {};
+
+	useEffect(() => {
+		if (project) {
+			const slaFiltedData = [];
+			const slaNewDate = (slaDate) => {
+				return new Date(slaDate)?.toLocaleDateString(
+					'en-US',
+					FORMAT_DATE
+				);
+			};
+
+			const slaRawData = {
+				current: {
+					slaDateEnd: slaNewDate(project.slaCurrentEndDate),
+					slaDateStart: slaNewDate(project.slaCurrentStartDate),
+					slaLabel: SLA_NAMES.current.toUpperCase(),
+					slaTitle: project.slaCurrent?.split(' ')[0],
+				},
+				expired: {
+					slaDateEnd: slaNewDate(project.slaExpiredEndDate),
+					slaDateStart: slaNewDate(project.slaExpiredStartDate),
+					slaLabel: SLA_NAMES.expired.toUpperCase(),
+					slaTitle: project.slaExpired?.split(' ')[0],
+				},
+				future: {
+					slaDateEnd: slaNewDate(project.slaFutureEndDate),
+					slaDateStart: slaNewDate(project.slaFutureStartDate),
+					slaLabel: SLA_NAMES.future.toUpperCase(),
+					slaTitle: project.slaFuture?.split(' ')[0],
+				},
+			};
+
+			if (
+				slaRawData.current.slaTitle === slaRawData.expired.slaTitle &&
+				slaRawData.current.slaTitle === slaRawData.future.slaTitle
+			) {
+				slaRawData.current.slaDateStart =
+					slaRawData.expired.slaDateStart;
+				slaRawData.current.slaDateEnd = slaRawData.future.slaDateEnd;
+				slaFiltedData.push(slaRawData.current);
+			} else if (
+				slaRawData.current.slaTitle === slaRawData.expired.slaTitle
+			) {
+				slaRawData.current.slaDateStart =
+					slaRawData.expired.slaDateStart;
+				slaFiltedData.push(slaRawData.current);
+				slaFiltedData.push(slaRawData.future);
+			} else if (
+				slaRawData.current.slaTitle === slaRawData.future.slaTitle
+			) {
+				slaRawData.current.slaDateEnd = slaRawData.future.slaDateEnd;
+				slaFiltedData.push(slaRawData.current);
+				slaFiltedData.push(slaRawData.expired);
+			} else {
+				slaFiltedData.push(slaRawData.current);
+				slaFiltedData.push(slaRawData.expired);
+				slaFiltedData.push(slaRawData.future);
+			}
+
+			setSlaData(slaFiltedData.filter((sla) => sla.slaTitle));
```

Comment:
This filter is necessary?

Reply:
Yes, because if don't have I will build SLA Card without information.